### ISSUE TITLE
Make multitask prediction consume QuerySeq evaluation grids

### ISSUE THREAD

**Issue by gkondas** [mmcdermott/EveryQuery-conditional#28]
Parent: #26  
Depends on: #27  
Blocks: #29.

## Base branch

Start only after #27 is complete, from its tip on the integration branch derived from `feat/prevalence-weighted-boundaries`.

## Goal

Make `EQ_predict_multitask` consume the consolidated `QuerySeqSchema` evaluation grid and score only the final real query in each row.

The prediction path must no longer require:

- `MultitaskBoundaryPytorchDataset`;
- packed `.labels.npy` evaluation targets;
- a multitask manifest;
- `eval_meta`;
- `eval_tasks.parquet`;
- constants imported from `sample_evaluation_multitask_sequences.py`;
- materialized `(B, K, V)` evaluation logits.

Do not change multitask training.

## 1. QuerySeq-to-multitask input adapter

For each real sequence row, map:

```text
q_durations        <- durations
q_bound_codes      <- bound_events, defaulting to NO_BOUND_INDEX
q_start_durations  <- start_durations, defaulting to 0.0
q_start_codes      <- start_events, defaulting to NO_BOUND_INDEX
condition_codes    <- queries[:-1]
condition_answers  <- answers[:-1]
scored_code        <- queries[-1]
label              <- answers[-1]
```

The final query is not teacher-forced into its own prediction.

Requirements:

- Support rows without the optional start or boundary columns.
- Validate query/start/bound codes against the checkpoint/cohort vocabulary before inference.
- Preserve the start/end sentinel invariants established in #27.
- Handle variable-length and padded batches correctly: `[:-1]` and `[-1]` refer to each row’s real sequence, never the padded batch width.
- Require at least one real query per row.
- Preserve right-padding and `q_mask` invariants expected by `ConditionalMultitaskARModel`.
- Do not allocate a fake `(B, K, V)` target tensor.

`ConditionalQueryPytorchDataset` must continue to reject active starts for the ordinary sequence-model path. The multitask predictor should use an explicit opt-in adapter/dataset path that tensorizes them.

Do not instantiate the checkpoint’s existing multitask datamodule unchanged: it expects the legacy metadata parquets, manifest, and packed-label sidecars. Reuse the checkpoint’s cohort/data settings while explicitly selecting the QuerySeq evaluation adapter.

## 2. Add target-only scoring

`ConditionalMultitaskARModel.forward` currently:

1. constructs window hidden states;
2. projects every window onto the entire tied vocabulary;
3. requires dense targets to validate width and compute BCE.

Refactor without changing the training result:

- Extract/reuse the hidden-state computation used by the existing forward pass.
- Add an inference path that selects the last real window per row.
- Select only that row’s final `scored_code`.
- Compute the scalar logit from the selected hidden state, the corresponding tied input-embedding row, and the corresponding `code_bias`.
- Return one logit/probability per input row.
- Do not enter the dense BCE path during prediction.

The existing all-vocabulary training forward, valid mask, target validation, logits, and loss must remain behaviorally unchanged.

### Numerical regression

On a small ordinary multitask batch:

1. run the existing full-vocabulary forward;
2. gather the final real window and selected code;
3. run the new target-only path;
4. assert the logits agree within floating-point tolerance.

Cover more than one selected code and more than one real sequence length.

## 3. Rewrite `predict_multitask.py`

Remove:

- `read_sidecars`;
- `align_sidecar`;
- `check_grid_coverage`;
- the old sidecar constants/import;
- dense-target gathering;
- `eval_meta_dir`.

Run sequential inference over the QuerySeq grid and verify that the dataloader yields exactly one prediction for every surviving input row. Continue to fail loudly if the tensorized cohort drops a grid subject.

Write one output row per input sequence containing at least:

```text
subject_id
prediction_time
queries
start_durations
start_events
durations
bound_events
answers
target_code
label
prob
```

Normalized default start/bound lists may be written even when their optional input columns were absent.

The complete lists identify the conditional task. The old evaluation-only `task_id`, `task_group`, `start_resolved`, `end_resolved`, and `window_days` sidecar fields are intentionally not recreated.

Update `src/every_query/predict/configs/predict_multitask.yaml`:

- `tasks_dir` points to `EQ_generate_evaluation_query_sequences`’s `eval` directory;
- remove `eval_meta_dir`;
- explain that one scalar final-query prediction is written per sequence row.

## 4. Tests

Rewrite `tests/multitask/test_predict_multitask_logic.py` to cover:

1. Missing start columns normalize to zeros/no-bound indices.
2. Active duration and event starts tensorize correctly through the explicit multitask opt-in.
3. Query, start, and end codes map to the right vocabulary indices.
4. Unknown codes fail before model inference.
5. Conditions are all and only the real queries before the final one.
6. The final code/label is selected per row under variable-length padding.
7. One-query sequences produce zero conditioning pairs.
8. Target-only logits equal gathered full-vocabulary logits.
9. The all-vocabulary training forward and loss remain unchanged.
10. Prediction output is row-aligned and contains one result per input row.
11. A grid subject absent from the tensorized cohort is rejected rather than silently dropped.
12. No evaluation manifest, packed labels, or eval-meta sidecar is needed.

Add an end-to-end test that generates a small QuerySeq grid with active starts and runs it through the multitask predictor far enough to verify the adapter/scoring/output contract.

## Completion

Run:

```bash
uv run pytest
uv run pytest -m 'slow or not slow'
uv run pre-commit run --files <changed files>
```

Do not run pre-commit over the full tree.

## Out of scope

- Changing the multitask training sampler or dataset.
- Removing the legacy evaluation generator; that happens only in the dependent cleanup issue.
- Adding active-start support to either ordinary sequence model.
- Adding ontology support to the multitask model.
